### PR TITLE
Sites/CPT: Refactor PublishMenu to accommodate custom post types

### PR DESCRIPTION
**Pull request by aduth**
This pull request seeks to refactor the `<PublishMenu />` component to better accommodate for custom post types. Notably, it:

- Requests and retrieves post types from global Redux state
- Uses accurate route paths and feature config for navigation to occur in development
- Preloads the custom post types bundle
- Simplifies logic for rendering menu items (general refactoring, ES6-ification)

__Testing instructions:__

The only observable difference should be that links for custom post types are no longer external in development environments.

1. Click My Sites in the master bar
2. Note that custom post types are still shown if applicable for site
 - Types shown should match that of master / production
3. For each custom post type, note that the link is not external*
4. Click the custom post type
5. Note that you are directed to the work-in-progress custom post types page
 - Should be presented with "Custom post types listing" text in content area

_*_ An external link includes an arrow icon: arrow

### CODE REVIEW

**Comment by aduth** [comment]
@timmyc : Rebased this and included stats tracking per #3875 in 7c41634 .

File: client/my-sites/sidebar/publish-menu.jsx [hunk 1]
```diff
@@ -124,84 +100,96 @@ var PublishMenu = React.createClass( {
...
-		customMenuItems = customPostTypes.map( function( postType ) {
+	getCustomMenuItems() {
+		const customPostTypes = omit( this.props.postTypes, [ 'post', 'page' ] );
+		return map( customPostTypes, function( postType ) {
 			return {
 				name: postType.name,
-				label: postType.labels ? postType.labels.menu_name : postType.label,
+				label: get( postType.labels, 'menu_name', postType.label ),
 				className: postType.name,
+				config: 'manage/custom-post-types',
```
Comment (timmyc) [inline client/my-sites/sidebar/publish-menu.jsx]: So this is still a dummy config correct?  Do you foresee us maybe having config flags on a per CPT basis?

File: client/my-sites/sidebar/publish-menu.jsx [hunk 2]
```diff
@@ -124,84 +100,96 @@ var PublishMenu = React.createClass( {
...
-			icon = 'folder';
-		} else if ( menuItem.name === 'jetpack-testimonial' ) {
-			icon = 'quote';
 		} else {
-			icon = 'custom-post-type';
+			preload = 'posts-custom';
+		}
+
+		let icon;
+		switch ( menuItem.name ) {
```
Comment (timmyc) [inline client/my-sites/sidebar/publish-menu.jsx]: This looks much nicer now - nice work.

**Comment by timmyc** [comment]
@aduth this is testing well for me.  I know this doesn't matter for production, but all I see when clicking a CPT is a blank page - no text

custom_post_type_ _wordpress_com

File: client/my-sites/sidebar/publish-menu.jsx (hunk 1 above)
Comment (aduth) [inline client/my-sites/sidebar/publish-menu.jsx]: >So this is still a dummy config correct?

This is now the [real config feature name](https://github.com/Automattic/wp-calypso/blob/6355759b2a7abe5b61f17d339d370d0973275177/config/development.json#L47). 

>Do you foresee us maybe having config flags on a per CPT basis?

Maybe at the point where we have CPT-specific layouts, but in the meantime, the layout should be consistent across post types, so one flag should be sufficient to cover them.

**Comment by aduth** [comment]
>but all I see when clicking a CPT is a blank page - no text

Since #4022 was merged, the pages should show a list of the posts you've created (published?) under that post type. It will probably show as blank if you have no posts - is that accurate?

**Comment by timmyc** [comment]
That is correct I have no posts for this post-type... just making sure it was okay.  Otherwise this looks good to go.  Verified stats are being tracked as well - thanks for adding that.

**Pull request merged**